ILSpy.ReadyToRun/ReadyToRunLanguage.cs

```diff
@@ -103,10 +103,84 @@ public override void WriteCommentLine(ITextOutput output, string comment)
+							&& varLoc.VariableLocation.VarLocType != VarLocType.VLT_STK_BYREF) {
+							output.WriteLine($"    Variable Number: {varLoc.VariableNumber}");
+							output.WriteLine($"    Start Offset: 0x{varLoc.StartOffset:X}");
+							output.WriteLine($"    End Offset: 0x{varLoc.EndOffset:X}");
+							output.WriteLine($"    Loc Type: {varLoc.VariableLocation.VarLocType}");
+							switch (varLoc.VariableLocation.VarLocType) {
+								case VarLocType.VLT_REG:
+								case VarLocType.VLT_REG_FP:
+								case VarLocType.VLT_REG_BYREF:
+									output.WriteLine($"    Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									break;
+								case VarLocType.VLT_STK:
+								case VarLocType.VLT_STK_BYREF:
+									output.WriteLine($"    Base Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									output.WriteLine($"    Stack Offset: {varLoc.VariableLocation.Data2}");
+									break;
+								case VarLocType.VLT_REG_REG:
+									output.WriteLine($"    Register 1: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									output.WriteLine($"    Register 2: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data2)}");
+									break;
+								case VarLocType.VLT_REG_STK:
+									output.WriteLine($"    Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									output.WriteLine($"    Base Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data2)}");
+									output.WriteLine($"    Stack Offset: {varLoc.VariableLocation.Data3}");
+									break;
+								case VarLocType.VLT_STK_REG:
+									output.WriteLine($"    Stack Offset: {varLoc.VariableLocation.Data1}");
+									output.WriteLine($"    Base Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data2)}");
+									output.WriteLine($"    Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data3)}");
+									break;
+								case VarLocType.VLT_STK2:
+									output.WriteLine($"    Base Register: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									output.WriteLine($"    Stack Offset: {varLoc.VariableLocation.Data2}");
+									break;
+								case VarLocType.VLT_FPSTK:
+									output.WriteLine($"    Offset: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									break;
+								case VarLocType.VLT_FIXED_VA:
+									output.WriteLine($"    Offset: {DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varLoc.VariableLocation.Data1)}");
+									break;
+								default:
+									throw new BadImageFormatException("Unexpected variable type");
```

Comment:
> Can we print a user-friendly message instead?

With my previous comment I was trying to suggest that you probably should use `output.WriteLine()` or similar instead of throwing an exception that effectively hides all the other output. Users might be exploring assemblies that are somewhat crippled. We still should try a best-effort decompilation and display as much information as possible. However, it's up to you.

```suggestion
									output.WriteLine("WRN: Unexpected variable location type");
```